### PR TITLE
Add support for NXP S32 EMIOS PWM

### PR DESCRIPTION
**Pull request by Dat-NguyenDuy**
This adds bare-metal driver for EMIOS on S32K3 devices for PWM operation

### CODE REVIEW

**Comment by manuargue** [comment]
@dleach02 @mmahadevan108 please help us to get reviewers assigned on this pr, thanks

**Review by mmahadevan108 — APPROVED** [review]

**Comment by jahedul-haque** [comment]
Please add a License File to the repo

**Comment by mmahadevan108** [comment]
@manuargue, do the changes look fine to you?

**Comment by DerekSnell** [comment]
> Please add a License File to the repo

Hi @jahedul-haque ,
Thank you for raising this. We created a new issue https://github.com/zephyrproject-rtos/zephyr/issues/61514 to track this, and NXP will be updating the HAL soon with a license file.

File: s32/drivers/s32k3/BaseNXP/include/Compiler.h [hunk 1]
```diff
@@ -483,18 +483,27 @@ extern "C"{
...
+
+#if defined(CONFIG_NOCACHE_MEMORY)
 #define VAR_SEC_NOCACHE(name) __attribute__((section(".nocache." STRINGIFY(name))))
 #else
 #define VAR_SEC_NOCACHE(name)
-#endif /* CONFIG_NOCACHE_MEMORY */
+#endif
+
+#if defined(CONFIG_NOCACHE_LOAD_MEMORY)
+#define VAR_SEC_NOCACHE_LOAD(name) __attribute__((section(".nocache_load." STRINGIFY(name))))
```
Comment (danieldegrasse) [inline s32/drivers/s32k3/BaseNXP/include/Compiler.h:503]: As I mentioned in the Zephyr PR, I do not think we should be creating an SOC specific Kconfig for loadable noncached memory. This feature is (at a minimum) useful for all Cortex-M7 SOCs

**Comment by manuargue** [comment]
> @manuargue, do the changes look fine to you?

LGTM but I raised this comment here https://github.com/zephyrproject-rtos/zephyr/pull/61310#discussion_r1288210040 which also Daniel seems to agree it would be a nice generic addition to have

File: s32/drivers/s32k3/BaseNXP/include/Compiler.h (hunk 1 above)
Comment (Dat-NguyenDuy) [reply s32/drivers/s32k3/BaseNXP/include/Compiler.h]: When i revisited the bare-metal driver and had a discussion with the engineer from HAL team, he said that actually the variable is supposed and can be put in to a read only region. In the mean time, added a patch here: https://github.com/zephyrproject-rtos/hal_nxp/commit/25cc567e5c832413060a63e3c1bb5c76338f18c9

**Review by bperseghetti — APPROVED** [review]

**Pull request merged**